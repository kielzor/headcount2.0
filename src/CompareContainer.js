import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const CompareContainer = ({createAverage, compared, evaluateCompareCard}) => {
  const displayCards = compared.map((cardData, i) =>
    <Card 
      data={cardData}
      evaluateCompareCard={evaluateCompareCard}
      key={`${cardData} + ${i}`}
    />
  );

  return (
    <div className='compare-box'>
      <div className='empty-compare'>{displayCards[0]}</div>
      <div className='compare-text'>
        {compared.length === 2 && <h4 className='compare-header'>Compared Locations</h4>}
        {compared.length === 2 && <h4 className='compare-num'>Averages</h4>}
        {compared.length === 2 && <h4 className='main-average'>{createAverage()[2]}</h4>}
        {compared.length === 1 && <h4 className='compare-header'>Choose another location</h4>}
        <div className='single-avgs'>
          {compared.length === 2 && <h4>{createAverage()[0]}</h4>}
          {compared.length === 2 && <h4>{createAverage()[1]}</h4>}
        </div>
      </div>
      <div className='empty-compare'>{displayCards[1]}</div>
    </div>
  );
};

CompareContainer.propTypes = {
  compared: PropTypes.array,
  evaluateCompareCard: PropTypes.func,
  district: PropTypes.object
};

export default CompareContainer;