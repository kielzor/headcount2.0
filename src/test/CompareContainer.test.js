import React from 'react'
import { shallow, mount } from 'enzyme'
import CompareContainer from '../CompareContainer'

describe('CompareContainer', () => {
  let wrapper
  let mockData
  let mockCompare

  beforeEach(() => {
    mockData = []
    const mockUpdate = jest.fn()
    wrapper = shallow(<CompareContainer 
                      compared={mockData}
                      comparedArr={mockCompare}/>)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should return information to the card', () => {
    expect(wrapper.find('div').length).toEqual(5)
    expect(wrapper.find('h4').length).toEqual(1)
  })
})