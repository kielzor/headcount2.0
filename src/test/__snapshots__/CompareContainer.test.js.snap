// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer
    compared={Array []}
    comparedArr={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="empty-compare"
/>,
        <div
          className="compare-text"
>
          <h4
                    className="compare-header"
          >
                    Compare Locations
          </h4>
          <div
                    className="single-avgs"
          >
                    
          </div>
</div>,
        <div
          className="empty-compare"
/>,
      ],
      "className": "compare-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "empty-compare",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="compare-header"
>
              Compare Locations
</h4>,
            false,
            false,
            <div
              className="single-avgs"
>
              
</div>,
          ],
          "className": "compare-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Compare Locations",
              "className": "compare-header",
            },
            "ref": null,
            "rendered": "Compare Locations",
            "type": "h4",
          },
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                false,
                false,
              ],
              "className": "single-avgs",
            },
            "ref": null,
            "rendered": Array [
              false,
              false,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "empty-compare",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="empty-compare"
/>,
          <div
            className="compare-text"
>
            <h4
                        className="compare-header"
            >
                        Compare Locations
            </h4>
            <div
                        className="single-avgs"
            >
                        
            </div>
</div>,
          <div
            className="empty-compare"
/>,
        ],
        "className": "compare-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "empty-compare",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="compare-header"
>
                Compare Locations
</h4>,
              false,
              false,
              <div
                className="single-avgs"
>
                
</div>,
            ],
            "className": "compare-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Compare Locations",
                "className": "compare-header",
              },
              "ref": null,
              "rendered": "Compare Locations",
              "type": "h4",
            },
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  false,
                  false,
                ],
                "className": "single-avgs",
              },
              "ref": null,
              "rendered": Array [
                false,
                false,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "empty-compare",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
